src/SearchExpression.h

```diff
@@ -62,6 +62,9 @@ namespace VDMS {
         /// The conjunctions of property predicates
         std::vector<PMGD::PropertyPredicate> _predicates;
 
+        /// The conjunctions of property predicates for edges
+        std::vector<PMGD::PropertyPredicate> _e_predicates;
+
```

Comment:
I suggest renaming these to be more clear: _predicates -> _node_predicates, and _e_predicates -> _edge_predicates.